server/gokbg3/grails-app/services/com/k_int/ESSearchService.groovy

```diff
@@ -423,99 +422,86 @@ class ESSearchService{
     log.debug("Processing platform value ${val} .. ")
   }
 
+
+  /**
+   * scroll : Get large amounts of data from the Elasticsearch index --
+   * @param params : Elasticsearch query params
+   * @return chunks of scrollSize data sets. In case the answer's size is smaller than scrollSize,
+   *         then the end of scrolling is reached.
+   **/
+  def scroll(params) throws Exception{
+
+    int scrollSize = 5000
+    def result = ["result" : "OK", "scrollSize" : scrollSize]
+    def esClient = ESWrapperService.getClient()
+
+    QueryBuilder query = (!params.component_type) ?
+        QueryBuilders.matchAllQuery() :
+        QueryBuilders.matchQuery("componentType", params.component_type)
+    // TODO: alternative query builders for scroll searches with q
+
+    ActionFuture<SearchResponse> response
+    if (!params.scrollId){
+      SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder()
+      searchSourceBuilder.query(query)
+      searchSourceBuilder.size(scrollSize)
+
+      SearchRequest searchRequest = new SearchRequest(grailsApplication.config.gokb.es.index)
+      searchRequest.scroll("1m")
+      searchRequest.source(searchSourceBuilder)
+      // ... set scroll interval to 1 minute
+      response = esClient.search(searchRequest)
+    }
+    else{
+      SearchScrollRequest scrollRequest = new SearchScrollRequest(params.scrollId)
+      scrollRequest.scroll("1m")
+      response = esClient.searchScroll(scrollRequest)
+    }
+    log.debug("scrollId : " + response.actionGet().getScrollId())
+    result.scrollId = response.actionGet().getScrollId()
+    SearchHit[] searchHits = response.actionGet().getHits().getHits()
+    result.records = []
+    for (SearchHit hit in searchHits){
+      result.records << hit.getSourceAsString()
```

Comment:
Why do we want the records as strings instead of objects?

Reply:
@hornmo True. Now being returned as a map.